From fc44453a82a59e5e825b2ee54f93126b559eb038 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Tue, 14 May 2024 17:05:34 +0800
Subject: [PATCH 092/137] mtk: hostapd: make 'band_idx' a mandatory
 configuration option

If 'band_idx' is not set, its default value is 0, which makes operations
that use band_idx always be applied on phy0.
(for example, pp_mode)

This commit gives 'band_idx' a default value 255, and returns error if
the value remains 255 after the config is read.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ap_config.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index 8b98d6170..95100b25c 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -313,6 +313,7 @@ struct hostapd_config * hostapd_config_defaults(void)
 	conf->ibf_enable = IBF_DEFAULT_ENABLE;
 	conf->amsdu = 1;
 	conf->pp_mode = PP_DISABLE;
+	conf->band_idx = 255;
 
 	hostapd_set_and_check_bw320_offset(conf, 0);
 
@@ -1631,6 +1632,11 @@ int hostapd_config_check(struct hostapd_config *conf, int full_config)
 		return -1;
 	}
 
+	if (full_config && conf->band_idx == 255) {
+		wpa_printf(MSG_ERROR, "band_idx is required");
+		return -1;
+	}
+
 	for (i = 0; i < conf->num_bss; i++) {
 		if (hostapd_config_check_bss(conf->bss[i], conf, full_config))
 			return -1;
-- 
2.18.0

